src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -995,20 +1004,142 @@ static inline JSC::EncodedJSValue jsBufferPrototypeFunction_fillBody(JSC::JSGlob
     }
 }
 
-static inline JSC::EncodedJSValue jsBufferPrototypeFunction_includesBody(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSBuffer>::ClassParameter castedThis)
+static int indexOf(const uint8_t* thisPtr, size_t thisLength, const uint8_t* valuePtr, size_t valueLength, int byteOffset)
+{
+    if (thisLength < valueLength + byteOffset)
+        return -1;
+    auto start = thisPtr + byteOffset;
+    auto end = thisPtr + thisLength;
+    auto it = std::search(start, end, std::boyer_moore_searcher(valuePtr, valuePtr + valueLength));
+    if (it != end) {
+        return it - thisPtr;
+    }
+    return -1;
+}
+
+static int lastIndexOf(const uint8_t* thisPtr, size_t thisLength, const uint8_t* valuePtr, size_t valueLength, int byteOffset)
+{
+    auto start = thisPtr;
+    auto end = thisPtr + std::min(thisLength, byteOffset + valueLength);
+    auto it = std::find_end(start, end, valuePtr, valuePtr + valueLength);
+    if (it != end) {
+        return it - thisPtr;
+    }
+    return -1;
+}
+
+static int indexOf(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSBuffer>::ClassParameter castedThis, bool last)
 {
     auto& vm = JSC::getVM(lexicalGlobalObject);
-    return JSC::JSValue::encode(jsUndefined());
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    if (callFrame->argumentCount() < 1) {
+        throwVMError(lexicalGlobalObject, scope, createNotEnoughArgumentsError(lexicalGlobalObject));
+        return JSValue::encode(jsUndefined());
+    }
+
+    auto value = callFrame->uncheckedArgument(0);
+    WebCore::BufferEncodingType encoding = WebCore::BufferEncodingType::utf8;
+
+    int length = static_cast<int>(castedThis->byteLength());
+    const uint8_t* typedVector = castedThis->typedVector();
+
+    int byteOffset = last ? length - 1 : 0;
+
+    if (callFrame->argumentCount() > 1) {
+        auto byteOffset_ = callFrame->uncheckedArgument(1).toNumber(lexicalGlobalObject);
```

Comment:
```suggestion
         byteOffset = callFrame->uncheckedArgument(0).toTypedArrayIndex(globalObject, "start");
        RETURN_IF_EXCEPTION(scope, encodedJSValue());
```

Reply:
This will let you delete the code checking for nan / isinf etc

Reply:
@Jarred-Sumner It seems that `toTypedArrayIndex` cannot be negative:
https://github.com/oven-sh/WebKit/blob/94c06efb6ed476047b2ed1532040943c72ae1502/Source/JavaScriptCore/runtime/JSCJSValueInlines.h#L92